### PR TITLE
RSDK-5607 - Cli Viam module build local

### PR DESCRIPTION
**Pull request by zaporter-work**
Ticket: https://viam.atlassian.net/browse/RSDK-5607

Questions for reviewers:
- Have I understood the module build process correctly? It runs the setup command and then runs the build command.

### CODE REVIEW

**Review by abe-winter — APPROVED** [review]

File: cli/module_build.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	}
+	if manifest.Build.Build != "" {
+		infof(c.App.Writer, "Starting build step: %q", manifest.Build.Build)
+		processConfig.Args = []string{"-c", manifest.Build.Build}
+		proc := pexec.NewManagedProcess(processConfig, logger.AsZap())
+		if err = proc.Start(c.Context); err != nil {
+			return err
+		}
+	} else {
+		return errors.New("the build command requires a non-empty build step")
```
[the remark this answers is not in the document]
Comment (zaporter-work) [reply cli/module_build.go:49]: Good point. Will fix.

File: cli/module_registry.go [hunk 2]
```diff
@@ -110,6 +126,7 @@ func CreateModuleAction(c *cli.Context) error {
 		Models: []ModuleComponent{
 			{},
 		},
+		Build: defaultBuildInfo,
```
[the remark this answers is not in the document]
Comment (zaporter-work) [reply cli/module_registry.go:129]: I think auto-populating them in the create step will lead to greater discoverability. Auto-defaulting to hidden values seems like increased cognitive load (ex there will be comments in peoples makefiles that says:
```
# This is the target from the build section of our meta.json
module.tar.gz:
   ... 
```
) and having it actually match the meta.json seems ideal.



I am happy to be wrong about this.

[the remark this answers is not in the document]
Comment (abe-winter) [reply cli/module_registry.go:129]: nope works for me

File: cli/module_build.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (zaporter-work) [reply cli/module_build.go]: Fixed and resolving.

File: cli/module_registry.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (zaporter-work) [reply cli/module_registry.go:129]: Ahh. I have realized one problem though... we shouldn't auto-populate until all the build commands are ready. Oops

[the remark this answers is not in the document]
Comment (zaporter-work) [reply cli/module_registry.go]: Commented out auto-populating until we are ready with the other commands

[the remark this answers is not in the document]
Comment (abe-winter) [reply cli/module_registry.go]: ah don't put them in while this is still hidden you mean, :+1:

**Pull request merged**